common/notifications.ts

```diff
@@ -2,12 +2,49 @@ export enum NotificationType {
   EmailVerification = 'emailVerification',
 }
 
-export interface EmailVerificationNotification {
+export interface BaseNotification {
+  /**
+   * A string that is unique if the notification is unique. For locally-generated notifications
+   * that should never show duplicates, this might be a constant string. For server-generated ones
+   * that have multiple of the same type (e.g. chat notifications, party invites, etc.) these are
+   * generated by the server.
+   */
+  readonly id: string
+  readonly type: NotificationType
+  readonly read: boolean
+  readonly createdAt: number
```

Comment:
This could definitely use a comment (honestly probably just comment all the fields here)